types/mydecimal_benchmark_test.go

```diff
@@ -70,43 +83,32 @@ func BenchmarkRound(b *testing.B) {
 	}
 }
 
-func BenchmarkToFloat64(b *testing.B) {
-	b.StopTimer()
-	tests := []struct {
-		input    string
-		inputDec MyDecimal
-	}{
-		{input: "123456789.987654321"},
-		{input: "15.1"},
-		{input: "15.5"},
-		{input: "15.9"},
-		{input: "-15.1"},
-		{input: "-15.5"},
-		{input: "-15.9"},
-		{input: "15.1"},
-		{input: "-15.1"},
-		{input: "15.17"},
-		{input: "15.4"},
-		{input: "-15.4"},
-		{input: "5.4"},
-		{input: ".999"},
-		{input: "999999999"},
+func BenchmarkToFloat64New(b *testing.B) {
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		for j := 0; j < numTestDec; j++ {
+			f, _ := testDec[j].ToFloat64()
+			_ = f
+		}
 	}
+}
 
-	for i := 0; i < len(tests); i++ {
-		err := tests[i].inputDec.FromString([]byte(tests[i].input))
-		if err != nil {
-			b.Fatal(err)
+func BenchmarkToFloat64Old(b *testing.B) {
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		for j := 0; j < numTestDec; j++ {
+			f, _ := strconv.ParseFloat(testDec[j].String(), 64)
+			_ = f
 		}
 	}
+}
 
-	b.StartTimer()
-	for n := 0; n < b.N; n++ {
-		for i := 0; i < len(tests); i++ {
-			_, err := tests[i].inputDec.ToFloat64()
-			if err != nil {
-				b.Fatal(err)
-			}
-		}
+func init() {
```

Comment:
Use init function is not good.   If i want to benchmark other function like BenchmarkRound, it shouldn't init testDec...
Use a normal function and call it in Benchmark function is ok

Reply:
Because I want to compare the benchmark of the new and old methods. So I gen the same workload for them and gen code should run before them.

Reply:
Using a normal function will also gen the same workload, I think....

Reply:
OK. That's right. I have modified it.